server/test/unit/server/webservices/views/test_repositories.py

```diff
@@ -0,0 +1,1977 @@
+    @mock.patch('pulp.server.webservices.views.repositories.reverse')
+    @mock.patch('pulp.server.webservices.views.repositories.manager_factory')
+    def test_post_repos_only_id(self, mock_factory, mock_reverse, mock_redir, mock_resp):
+        """
+        Create a repo using the minimal body.
+        """
+
+        mock_request = mock.MagicMock()
+        mock_request.body = json.dumps({'id': 'mock_repo'})
+        mock_reverse.return_value = '/mock/path/'
+        mock_repo_manager = mock_factory.repo_manager.return_value
+        mock_repo_manager.create_and_configure_repo.return_value = {'id': 'mock_repo'}
+
+        repos_view = ReposView()
+        response = repos_view.post(mock_request)
+
+        expected_content = {'id': 'mock_repo', '_href': '/mock/path/'}
+        mock_resp.assert_called_once_with(expected_content)
+        mock_redir.assert_called_once_with(mock_resp.return_value, expected_content['_href'])
+        self.assertTrue(response is mock_redir.return_value)
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_CREATE())
+    @mock.patch(
+        'pulp.server.webservices.views.repositories.generate_json_response_with_pulp_encoder')
+    @mock.patch('pulp.server.webservices.views.repositories.generate_redirect_response')
+    @mock.patch('pulp.server.webservices.views.repositories.reverse')
+    @mock.patch('pulp.server.webservices.views.repositories.manager_factory')
+    def test_post_repos_all_fields(self, mock_factory, mock_reverse, mock_redir, mock_resp):
+        """
+        Create a repo using all allowed fields.
+        """
+
+        mock_repo = {
+            'id': 'mock_repo', 'display_name': 'mock_display', 'description': 'mock_desciption',
+            'notes': 'mock_notes', 'importer_type_id': 'mock_importer', 'distributors': ['dist1']
+        }
+        mock_request = mock.MagicMock()
+        mock_request.body = json.dumps(mock_repo)
+        mock_reverse.return_value = '/mock/path/'
+        mock_repo_manager = mock_factory.repo_manager.return_value
+        mock_repo_manager.create_and_configure_repo.return_value = mock_repo
+
+        repos_view = ReposView()
+        response = repos_view.post(mock_request)
+
+        mock_repo['_href'] = '/mock/path/'
+        mock_resp.assert_called_once_with(mock_repo)
+        mock_redir.assert_called_once_with(mock_resp.return_value, mock_repo['_href'])
+        self.assertTrue(response is mock_redir.return_value)
+
+
+class TestRepoResourceView(unittest.TestCase):
+    """
+    Tests for RepoResoureceView.
+    """
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_READ())
+    @mock.patch(
+        'pulp.server.webservices.views.repositories.generate_json_response_with_pulp_encoder')
+    @mock.patch('pulp.server.webservices.views.repositories.manager_factory')
```

Comment:
ah i see, anyway no harm would be done here as last_unit_added is None same as last_unit_removed